### PR TITLE
ES|QL refactor PositionToXContent

### PR DESCRIPTION
**Pull request by ChrisHegarty**
This commit is pure refactoring - to extract PositionToXContent out of ColumnInfo.

PositionToXContent adds utility support for transforming column info/data to xcontent, it is not intrinsically part of the ColumnInfo itself. This refactoring is part of a small set of precursor changes required to expose a subset of the transport API to other parts of Elasticsearch, see #106244.

### CODE REVIEW

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/PositionToXContent.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+import org.elasticsearch.xpack.versionfield.Version;
+
+import java.io.IOException;
+
+import static org.elasticsearch.xpack.ql.util.DateUtils.UTC_DATE_TIME_FORMATTER;
+import static org.elasticsearch.xpack.ql.util.NumericUtils.unsignedLongAsNumber;
+import static org.elasticsearch.xpack.ql.util.SpatialCoordinateTypes.CARTESIAN;
+import static org.elasticsearch.xpack.ql.util.SpatialCoordinateTypes.GEO;
+
+public abstract class PositionToXContent {
```
Comment (costin) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/PositionToXContent.java:35]: If it's not used outside the package, make it package protected.

**Review by costin — APPROVED** [review]
👍

**Review by nik9000 — APPROVED** [review]

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Pull request merged**